src/wkt/tests/message_with_f32.rs

```diff
@@ -0,0 +1,133 @@
+        pub repeated: Vec<f32>,
+        #[serde_as(as = "std::collections::HashMap<_, google_cloud_wkt::internal::F32>")]
+        pub hashmap: std::collections::HashMap<String, f32>,
+    }
+
+    #[test_case(9876.5, 9876.5)]
+    #[test_case(f32::INFINITY, "Infinity")]
+    #[test_case(f32::NEG_INFINITY, "-Infinity")]
+    #[test_case(f32::NAN, "NaN")]
+    fn test_singular<T>(input: f32, want: T) -> Result
+    where
+        T: serde::ser::Serialize,
+    {
+        let msg = MessageWithF32 {
+            singular: input,
+            ..Default::default()
+        };
+        let got = serde_json::to_value(&msg)?;
+        let want = json!({"singular": want, "repeated": [], "hashmap": {}});
+        assert_eq!(want, got);
+
+        let roundtrip = serde_json::from_value::<MessageWithF32>(got)?;
+        assert_float_eq(msg.singular, roundtrip.singular);
+        Ok(())
+    }
+
+    #[test_case(9876.5, 9876.5)]
+    #[test_case(f32::INFINITY, "Infinity")]
+    #[test_case(f32::NEG_INFINITY, "-Infinity")]
+    #[test_case(f32::NAN, "NaN")]
+    fn test_optional<T>(input: f32, want: T) -> Result
+    where
+        T: serde::ser::Serialize,
+    {
+        let msg = MessageWithF32 {
+            optional: Some(input),
+            ..Default::default()
+        };
+        let got = serde_json::to_value(&msg)?;
+        let want = json!({"singular": 0.0, "optional": want, "repeated": [], "hashmap": {}});
+        assert_eq!(want, got);
+
+        let roundtrip = serde_json::from_value::<MessageWithF32>(got)?;
+        assert_float_eq(msg.optional.unwrap(), roundtrip.optional.unwrap());
+        Ok(())
+    }
+
+    #[test]
+    fn test_repeated() -> Result {
+        let msg = MessageWithF32 {
+            repeated: vec![f32::INFINITY, f32::NEG_INFINITY, f32::NAN, 9876.5_f32],
+            ..Default::default()
+        };
+        let got = serde_json::to_value(&msg)?;
+        let want = json!({"singular": 0.0, "repeated": ["Infinity", "-Infinity", "NaN", 9876.5], "hashmap": {}});
+        assert_eq!(want, got);
+
+        let roundtrip = serde_json::from_value::<MessageWithF32>(got)?;
+        for (roundtrip, msg) in roundtrip.repeated.iter().zip(msg.repeated.iter()) {
+            assert_float_eq(*roundtrip, *msg);
+        }
+        Ok(())
+    }
+
+    #[test]
+    fn test_hashmap() -> Result {
+        let mut hashmap = std::collections::HashMap::new();
+        hashmap.insert("number".to_string(), 9876.5);
+        hashmap.insert("inf".to_string(), f32::INFINITY);
+        hashmap.insert("-inf".to_string(), f32::NEG_INFINITY);
+        hashmap.insert("nan".to_string(), f32::NAN);
+
+        let msg = MessageWithF32 {
+            hashmap,
+            ..Default::default()
+        };
+
+        let got = serde_json::to_value(&msg)?;
+        let want = json!({"singular": 0.0, "repeated": [],"hashmap": {"number": 9876.5, "inf": "Infinity", "-inf": "-Infinity", "nan": "NaN"}});
```

Comment:
nit: consider addine newlines for readability:

```rs
        let want = json!({
            "singular": 0.0,
            "hashmap": {
                "number": 9876.5,
                "inf": "Infinity",
                "-inf": "-Infinity",
                "nan": "NaN"
            }
        });
```

(I am assuming we will remove the `repeated` field)

Reply:
Done.